qiskit/circuit/library/generalized_gates/mcmt.py

```diff
@@ -142,11 +143,20 @@ def _identify_gate(self, gate):
 
         return base_gate
 
-    def control(self, num_ctrl_qubits=1, label=None, ctrl_state=None):
+    def control(self, num_ctrl_qubits=1, label=None, ctrl_state=None, annotated=False):
         """Return the controlled version of the MCMT circuit."""
-        if ctrl_state is None:  # TODO add ctrl state implementation by adding X gates
-            return MCMT(self.gate, self.num_ctrl_qubits + num_ctrl_qubits, self.num_target_qubits)
-        return super().control(num_ctrl_qubits, label, ctrl_state)
+        if not annotated:
+            if ctrl_state is None:  # TODO add ctrl state implementation by adding X gates
+                gate = MCMT(
+                    self.gate, self.num_ctrl_qubits + num_ctrl_qubits, self.num_target_qubits
+                )
+            else:
+                gate = super().control(num_ctrl_qubits, label, ctrl_state, annotated=annotated)
+        else:
+            gate = AnnotatedOperation(
+                self, ControlModifier(num_ctrl_qubits=num_ctrl_qubits, ctrl_state=ctrl_state)
+            )
```

Comment:
I'd probably have done to save some code and reduce duplication:

```suggestion
        if not annotated and ctrl_state is None:
            gate = MCMT(
                self.gate, self.num_ctrl_qubits + num_ctrl_qubits, self.num_target_qubits
            )
        else:
            gate = super().control(num_ctrl_qubits, label, ctrl_state, annotated=annotated)
```

I think this applies pretty much everwhere.

Reply:
Oh, I absolutely love this suggestion. Not only this makes the code shorter and more readable, it also seems a more correct way to apply the control method. Changed across the board in [0783fbb](https://github.com/Qiskit/qiskit/pull/11433/commits/0783fbb6e898593c876b95639f1e9ad1b83c8e8c).